homeassistant/components/zeroconf/__init__.py

```diff
@@ -432,6 +432,7 @@ def async_service_update(
                     zeroconf, async_service_info, service_type, name
                 ),
                 name=f"zeroconf lookup {name}.{service_type}",
+                eager_start=False,
```

Comment:
zeroconf still has a bug where the listener can't be changed inside a callback because set size will change during iteration